include/no_os_eeprom.h

```diff
@@ -50,6 +50,8 @@
 /********************** Macros and Constants Definitions **********************/
 /******************************************************************************/
 
+#define EEPROM_MAX_NUMBER 4
```

Comment:
For Pwm/Timer/Uart it's pretty clear, they are hw peripherals that have a "device id" and having a static table with mutexes for each makes sense. But in the case of EEPROM and MDIO, it's not clear to me how to handle them. I would propose to remove the eeprom/mdio changes from this PR, think them through and add them later, separately. For me t's not even clear what an EEPROM device ID is, and that ID is used as a index in the mutex table. So I think we need to sort that out in the team first.

Reply:
removed